File: fabric/src/main/java/com/selfdot/cobblemontrainers/fabric/FabricPermissionValidator.java
package com.selfdot.cobblemontrainers.fabric;

import com.cobblemon.mod.common.api.permission.Permission;
import com.selfdot.cobblemontrainers.command.permission.PermissionValidator;
import me.lucko.fabric.api.permissions.v0.Permissions;
import net.minecraft.command.CommandSource;

public class FabricPermissionValidator implements PermissionValidator {

    @Override
    public boolean hasPermission(CommandSource source, Permission permission) {
        return Permissions.check(
            source, "selfdot." + permission.getLiteral(), permission.getLevel().getNumericalValue()
        );
    }

}

File: fabric/src/main/kotlin/com/selfdot/cobblemontrainers/fabric/FabricPermissionValidator.kt
package com.selfdot.cobblemontrainers.fabric

import com.cobblemon.mod.common.api.permission.Permission
import com.selfdot.cobblemontrainers.command.permission.PermissionValidator
import me.lucko.fabric.api.permissions.v0.Permissions
import net.minecraft.command.CommandSource

class FabricPermissionValidator : PermissionValidator {

    override fun hasPermission(source: CommandSource, permission: Permission) =
        Permissions.check(source, "selfdot." + permission.literal, permission.level.numericalValue)

}
